mediapipe/tasks/ios/text/core/sources/MPPBaseTextTaskApi.h

```diff
@@ -0,0 +1,48 @@
+/* Copyright 2022 The TensorFlow Authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ ==============================================================================*/
+#import <Foundation/Foundation.h>
+
+#include "mediapipe/framework/calculator.pb.h"
+#include "mediapipe/tasks/cc/core/task_runner.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ * The base class of the user-facing iOS mediapipe text task api classes.
+ */
+NS_SWIFT_NAME(BaseTextTaskApi)
+@interface MPPBaseTextTaskApi : NSObject {
```

Comment:
@schmidt-sebastian @khanhlvg 
This is the base class for all text APIs. Please note that C++ header files are included here.  These headers and any headers transitively imported by them will have to be part of the iOS framework we build for distribution. According to me this is not a good design as consumers of the API will be exposed to many C++ headers. Also I am yet to check the feasibility of this design when building CocoaPods. 

**My Suggestion:**
 Instead of extending the individual tasks from a  base task api, we can create a base task util class which will encapsulate the c++ task runner and all the common functionality needed. The utils will only be included in the .mm file. So we don't have to expose the C++ headers.

Reply:
Can you elaborate on this? The headers will be part of the build process, but does the user have to consume them in order to use our iOS API?